### ISSUE TITLE
Expose SonarQube server URL as a build parameter

### ISSUE THREAD

**Issue by ogerovich** [JetBrains/TeamCity.SonarQubePlugin#70]
There is already %runParam.sonarServer%, but it's some kind of unique identifier. Please provide something like %runParam.sonarServerUrl%. This way I could pass it as a parameter within a build step to my custom script.

**Comment by Linfar** [comment]
Yes, sonarServer is UUID of a connection used in a certain build step. When build is started a number of additional parameters are resolved basing on this UUID, including `sonar.host.url` with the URL. However due to current implementation these parameters are set only for starting build and cannot be accessed on server side. 

Could you please describe how do you get access to runner parameters - where `runParam` prefix comes from?

**Comment by ogerovich** [comment]
When I was testing my script earlier, I was using SonarQube plugin 2017.1-28. I used a [Python](https://github.com/leo-from-spb/teamcity-python) build step. I started typing `%sonar` in command line arguments text box, and a few matches appeared. `%runParam.sonarServer%` was among them. When I tried to run the build, I discovered that it's the UUID. I have since updated SonarQube plugin to version 2017.1-30, and auto-complete no longer shows any matches for `%sonar`. Was this disabled between 2017.1-28 and 2017.1-30?

It does not matter to me how the parameter is exposed as long as I can use it either in a build step or to define another build parameter. Ideally, auto-complete works, but this is not a requirement.

**Comment by Linfar** [comment]
`%runParam.sonarServer%` isn't a parameter provided by SonarQube plugin (and has never been). AFAIK default TeamCity functionality doesn't provide `runParam.` prefix neither. Moreover runner parameters can exist only during executing step where these parameters are defined and mean nothing in other steps. 

So I cannot easily provide such parameter from the plugin - eg there can be 2 different Sonar Runner usages with different servers during one build. 

MSBuild Scanner seems to be easier here since usually there is only one simultaneous Start task (there can be subsequent 2 start-finish tasks during the build but I really doubt anyone will ever intersect them) and server is used in the Start task. So probably I can set parameter value between start and finish. 

I'll try to find some way to provide SonarQube server urls. If you can describe how you'll use such parameter it could help me.
